### PR TITLE
Update OpenAPI Spec & Merge Files API

### PR DESCRIPTION
**Pull request by mgyucht**
## Changes
This PR adds recent changes from OpenAPI to the Java SDK. Notably, it adds support for the Files API. The integration test for this API can only be run in UC workspaces, so this PR also adds support for UC-only integration tests at the workspace- and account-level.

## Tests
- [x] New integration test for Files API to verify that it works as expected.

### CODE REVIEW

File: .codegen/impl.java.tmpl [hunk 1]
```diff
@@ -26,18 +28,36 @@ class {{.PascalName}}Impl implements {{.PascalName}}Service {
...
+apiClient.{{.Verb}}(path
+  {{- if .Request}}, {{ template "request-param" .}}{{end}}
+  , {{ if not .Response -}}Void
+  {{- else if .IsResponseByteStream}}InputStream
+  {{- else}}{{template "type" .Response}}{{- end -}}.class
+  , headers);
+{{- end }}
+
+{{ define "headers" -}}
+    Map<String, String> headers = new HashMap<>();
```
Comment (nfx) [inline .codegen/impl.java.tmpl:55]: this results in the unnecessary diff and headers variable for literally every single method. surround it with `if` and make sure there's no diff in the unaffected methods
```suggestion
    {{if .FixedRequestHeaders}}
    ... what you have here
    {{end}}
```

File: .codegen/impl.java.tmpl [hunk 2]
```diff
@@ -26,18 +28,36 @@ class {{.PascalName}}Impl implements {{.PascalName}}Service {
...
-      {{- else -}} return apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-                        {{- else if .Response.MapValue }}Map
-                        {{- else}}{{template "type" .Response}}
-                        {{- end -}}{{else}}, Void{{end}}.class);{{end}}
-    {{- else -}}apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-      {{- else if .Response.MapValue }}Map
-      {{- else}}{{template "type" .Response}}
-      {{- end -}}{{else}}, Void{{end}}.class);
-    {{end}}
+    {{ template "headers" . -}}
```
Comment (nfx) [inline .codegen/impl.java.tmpl:31]:
```suggestion
    {{- template "headers" . -}}
```

to cut whitespace before

File: .codegen/impl.java.tmpl [hunk 3]
```diff
@@ -26,18 +28,36 @@ class {{.PascalName}}Impl implements {{.PascalName}}Service {
...
+    {{ if not .Response -}}
+    {{ template "api-call" . }}
+    {{- else if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class, headers);
+    {{- else if .Response.MapValue -}} return apiClient.getStringMap(path, {{ template "request-param" .}}, headers);
+    {{- else if .IsResponseByteStream -}}
+    InputStream response = {{ template "api-call" . }}
```
Comment (nfx) [inline .codegen/impl.java.tmpl:37]: why not returning `InputStream` as a result type? it makes sense to return a wrapping response type only if there's more than one field in it, otherwise it's confusing.

File: .codegen/impl.java.tmpl [hunk 4]
```diff
@@ -26,18 +28,36 @@ class {{.PascalName}}Impl implements {{.PascalName}}Service {
...
-                        {{- end -}}{{else}}, Void{{end}}.class);{{end}}
-    {{- else -}}apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-      {{- else if .Response.MapValue }}Map
-      {{- else}}{{template "type" .Response}}
-      {{- end -}}{{else}}, Void{{end}}.class);
-    {{end}}
+    {{ template "headers" . -}}
+    {{ if not .Response -}}
+    {{ template "api-call" . }}
+    {{- else if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class, headers);
```
Comment (nfx) [inline .codegen/impl.java.tmpl:34]: why this cannot be solved with overloading?
```suggestion
    {{- else if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class{{.HasRequestReaders}}, headers{{end}});
```

**Review by nfx — COMMENTED** [review]
the diff seems to be bigger than necessary just for the addition of request headers.

File: databricks-sdk-java/src/main/java/com/databricks/sdk/core/commons/CommonsHttpClient.java [hunk 5]
```diff
@@ -84,8 +84,7 @@ private Response computeResponse(Request in, CloseableHttpResponse response) thr
     boolean streamResponse =
         in.getHeaders().containsKey("Accept")
             && !APPLICATION_JSON.getMimeType().equals(in.getHeaders().get("Accept"))
-            && hs.containsKey("Content-Type")
```
Comment (mgyucht) [inline databricks-sdk-java/src/main/java/com/databricks/sdk/core/commons/CommonsHttpClient.java:87]: This is needed because header keys are not case-sensitive (esp. if response is HTTP 2.0, which enforces header keys to be lower case).

**Review by tanmay-db — APPROVED** [review]
Thanks for going over the PR for review. LGTM
